sympy/matrices/common.py

```diff
@@ -2649,7 +2650,7 @@ def entry(i, j):
                 # Some matrices don't work with `sum` or `Add`
                 # They don't work with `sum` because `sum` tries to add `0`
                 # Fall back to a safe way to multiply if the `Add` fails.
-                return reduce(lambda a, b: a + b, vec)
+                return reduce(operator.add, vec)
```

Comment:
This should really be `Add(*items)`.

I am not sure if the comment above is still accurate. It refers to deprecated subclasses that can be removed now.

Reply:
This is in a try except. Add(*items) is in the try portion of the except, so I think this is intentional too. Could you double check if this what you intended? Or should the entire try except block just be removed?